src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -134,6 +142,92 @@ public Command stowTurret() {
+  }
+
+  /** Will calculate the current turret position and update encoders and motors off of it. */
+  public void zeroTurret() {
+    double lowGearPosition = lowGearCANcoder.getAbsolutePosition().getValue().doubleValue();
+    double highGearPosition = highGearCANcoder.getAbsolutePosition().getValue().doubleValue();
+    Rotation2d turretRotation = encoderPositionsToTurretRotation(lowGearPosition, highGearPosition);
+
+    lowGearCANcoder.setPosition(
+        turretRotation.getRotations() * Constants.TurretConstants.LOW_GEAR_CAN_CODER_RATIO);
+    highGearCANcoder.setPosition(
+        turretRotation.getRotations() * Constants.TurretConstants.HIGH_GEAR_CAN_CODER_RATIO);
+    turretMotor
+        .getEncoder()
+        .setPosition(
+            turretRotation.getRotations() * Constants.TurretConstants.TURRET_MOTOR_TURRET_RATIO);
+  }
+
+  /**
+   * Calculates the turret zero off of given encoder rotations.
+   *
+   * @param lowGearCANcoderPosition The low gear encoder position, as in the gear with a lower gear
+   *     ratio
+   * @param highGearCANcoderPosition The high gear encoder position, as in the gear with a higher
+   *     gear ratio
+   * @return the calculated turret rotation
+   */
+  public static Rotation2d encoderPositionsToTurretRotation(
+      double lowGearCANcoderPosition, double highGearCANcoderPosition) {
+
+    // This equation is based off of
+    // https://www.geeksforgeeks.org/implementation-of-chinese-remainder-theorem-inverse-modulo-based-implementation/
+    // It is accurate to with in 3.6 deg
+    int gearToothPosition =
+        ((int) (lowGearCANcoderPosition * Constants.TurretConstants.LOW_GEAR_CANCODER_TEETH)
+                    * HowdyMath.inverse_modulus(
+                        Constants.TurretConstants.HIGH_GEAR_CANCODER_TEETH,
+                        Constants.TurretConstants.LOW_GEAR_CANCODER_TEETH)
+                    * Constants.TurretConstants.HIGH_GEAR_CANCODER_TEETH
+                + (int)
+                        (highGearCANcoderPosition
+                            * Constants.TurretConstants.HIGH_GEAR_CANCODER_TEETH)
+                    * HowdyMath.inverse_modulus(
+                        Constants.TurretConstants.LOW_GEAR_CANCODER_TEETH,
+                        Constants.TurretConstants.HIGH_GEAR_CANCODER_TEETH)
+                    * Constants.TurretConstants.LOW_GEAR_CANCODER_TEETH)
+            % (Constants.TurretConstants.LOW_GEAR_CANCODER_TEETH
+                * Constants.TurretConstants.HIGH_GEAR_CANCODER_TEETH);
+
+    double roughRotation = gearToothPosition / (Constants.TurretConstants.TURRET_GEAR_TEETH + 0.0);
+    double lowGearCANCoderDivsionSize =
+        (Constants.TurretConstants.LOW_GEAR_CANCODER_TEETH + 0.0)
+            / Constants.TurretConstants.TURRET_GEAR_TEETH;
+    double highGearCANCoderDivsionSize =
+        (Constants.TurretConstants.HIGH_GEAR_CANCODER_TEETH + 0.0)
+            / Constants.TurretConstants.TURRET_GEAR_TEETH;
```

Comment:
This is just 1/<HIGH/LOW>_GEAR_CAN_CODER_RATIOs. Might be more convenient to just use those constants and swap your division/multiply signs.

Reply:
X_RATIO is used in multiple locations. Leaving it how it is currently the easiest route